Source/Tools/FEXLoader/LinuxSyscalls/SignalDelegator.cpp

```diff
@@ -1442,10 +1540,11 @@ namespace FEX::HLE {
        Handler.DefaultBehaviour == DEFAULT_TERM)) {
       // Reassign back to DFL and crash
       signal(Signal, SIG_DFL);
-      if (SigInfo->si_code != SI_KERNEL) {
+      if (SigInfo.si_code != SI_KERNEL) {
         // If the signal wasn't sent by the kernel then we need to reraise it.
         // This is necessary since returning from this signal handler now might just continue executing.
         // eg: If sent from tgkill then the signal gets dropped and returns.
+        Thread->CurrentFrame->State.DeferredSignalRefCount.Store(0);
```

Comment:
This is a fatal case that is happening. This being deferred or not actually doesn't matter anymore.
Removed because it would just be confusing that it does matter. The following tgkill line is going to kill the process anyway.